libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py

```diff
@@ -0,0 +1,372 @@
+                "Please install it with `pip install oracle_ads`."
+            ) from ex
+        if not values.get("auth", None):
+            values["auth"] = ads.common.auth.default_signer()
+        values["endpoint"] = get_from_dict_or_env(
+            values,
+            "endpoint",
+            "OCI_LLM_ENDPOINT",
+        )
+        return values
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        """Default parameters for the model."""
+        raise NotImplementedError
+
+    @property
+    def _identifying_params(self) -> Dict[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            **{"endpoint": self.endpoint},
+            **self._default_params,
+        }
+
+    def _construct_json_body(self, prompt: str, params: dict) -> dict:
+        """Constructs the request body as a dictionary (JSON)."""
+        raise NotImplementedError
+
+    def _invocation_params(self, stop: Optional[List[str]], **kwargs: Any) -> dict:
+        """Combines the invocation parameters with default parameters."""
+        params = self._default_params
+        if self.stop is not None and stop is not None:
+            raise ValueError("`stop` found in both the input and default params.")
+        elif self.stop is not None:
+            params["stop"] = self.stop
+        elif stop is not None:
+            params["stop"] = stop
+        else:
+            # Don't set "stop" in param as None. It should be a list.
+            params["stop"] = []
+
+        return {**params, **kwargs}
+
+    def _process_response(self, response_json: dict) -> str:
+        raise NotImplementedError
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> str:
+        """Call out to OCI Data Science Model Deployment endpoint.
+
+        Args:
+            prompt (str):
+                The prompt to pass into the model.
+            stop (List[str], Optional):
+                List of stop words to use when generating.
+            kwargs:
+                requests_kwargs:
+                    Additional ``**kwargs`` to pass to requests.post
+
+        Returns:
+            The string generated by the model.
+
+        Example:
+            .. code-block:: python
+
+                response = oci_md("Tell me a joke.")
+
+        """
+        requests_kwargs = kwargs.pop("requests_kwargs", {})
+        params = self._invocation_params(stop, **kwargs)
+        body = self._construct_json_body(prompt, params)
+        logger.info(f"LLM API Request:\n{prompt}")
+        response = self.send_request(
+            data=body, endpoint=self.endpoint, **requests_kwargs
+        )
+        completion = self._process_response(response)
+        logger.info(f"LLM API Completion:\n{completion}")
+        return completion
+
+    def send_request(
```

Comment:
Is this necessary for clients to know about? Could we rename to not be public?
```suggestion
    def _send_request(
```

Reply:
Sure. Changed to private method. Thanks!